app/scripts/modules/kubernetes/src/common/tolerations/TolerationsComponent.tsx

```diff
@@ -0,0 +1,194 @@
+import * as React from 'react';
+import { cloneDeep } from 'lodash';
+import Select, { Option } from 'react-select';
+import { Button } from 'react-bootstrap';
+import { BindAll } from 'lodash-decorators';
+import { HelpField } from '@spinnaker/core';
+
+export type IOperator = 'Exists' | 'Equal';
+export type IEffect = 'NoSchedule' | 'PreferNoSchedule' | 'NoExecute';
+
+export interface IToleration {
+  key: string;
+  operator: IOperator;
+  effect: IEffect;
+  value: string;
+  tolerationSeconds?: number;
+}
+
+export interface IKubernetesTolerationsProps {
+  tolerations: IToleration[];
+  onTolerationChange: (tolerations: IToleration[]) => void;
+}
+
+export interface IKubernetesTolerationsState {
+  tolerations: IToleration[];
+}
+
+@BindAll()
+export class KubernetesTolerations extends React.Component<IKubernetesTolerationsProps, IKubernetesTolerationsState> {
+  private static operators: Option[] = [{ value: 'Exists', label: 'Exists' }, { value: 'Equal', label: 'Equal' }];
+
+  private static effects: Option[] = [
+    { value: 'NoSchedule', label: 'NoSchedule' },
+    { value: 'PreferNoSchedule', label: 'PreferNoSchedule' },
+    { value: 'NoExecute', label: 'NoExecute' },
+  ];
+
+  public tolerations: IToleration[];
+
+  constructor(props: IKubernetesTolerationsProps) {
+    super(props);
+    this.state = this.getState(props);
+  }
+
+  private getState(props: IKubernetesTolerationsProps): IKubernetesTolerationsState {
```

Comment:
Since state will always just be `props.tolerations`, you probably don't even need state at all, then you can get rid of `componentWillReceiveProps` as well.